private/pkg/changelog/cmd/update-changelog/main.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// update-changelog supplies a cli that updates the CHANGELOG.md. The tool accepts two operations: "release" and "unrelease".
+// "update-changelog release" requires a filename argument (default CHANGELOG.md), a --version flag in the form vx.y.z, an optional --date flag. If no date is supplied the current date will be used.
+// "update-changelog unrelease" does not require any flags or arguments except for the optional filename and will add `Unreleased` sections to the changelog.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+)
+
+const semverRegex = `((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)`
+
+func main() {
```

Comment:
generally we'd do `if err := run(); err != nil` style, grep the codebase for examples